src/Pelagos/Bundle/AppBundle/Command/ColdStorageFlagCommand.php

```diff
@@ -0,0 +1,181 @@
+                    $newDatasetSubmissionIdGenerator = $metadata->idGenerator;
+                    if ($newDatasetSubmissionId !== null) {
+                        // Temporarily change the ID generator to AssignedGenerator.
+                        $metadata->setIdGenerator(new AssignedGenerator());
+                    }
+
+                    $entityManager->persist($newDatasetSubmission);
+                    $entityManager->flush($newDatasetSubmission);
+
+                    if ($newDatasetSubmissionId !== null) {
+                        $metadata->setIdGenerator($newDatasetSubmissionIdGenerator);
+                    }
+
+                    $dataset->setDatasetSubmission($newDatasetSubmission);
+                    $entityManager->persist($dataset);
+                    $entityManager->flush($dataset);
+
+                    // This is an ugly hack, having to re-persist, but there seems not to be a cleaner way...
+                    // Reset original status.
+                    try {
+                        $newDatasetSubmissionReflection = new \ReflectionClass($newDatasetSubmission);
+                        $statusReflection = $newDatasetSubmissionReflection->getProperty('status');
+                        $statusReflection->setACcessible(true);
+                        $statusReflection->setValue($newDatasetSubmission, $datasetSubmission->getStatus());
+                    } catch (\ReflectionException $exception) {
+                        throw new \ReflectionException('Reflection class failed ' . $exception->getMessage());
+                    }
+                    $entityManager->persist($newDatasetSubmission);
+                    $entityManager->flush($newDatasetSubmission);
+
+                    $output->writeln('Original DatasetSubmission Status: ' . $datasetSubmission->getStatus());
+                    $output->writeln('New submission ID is now: ' . $newDatasetSubmission->getId());
+
+                    $output->writeln('New DatasetSubmission Status referenced in dataset: '
+                        . $dataset->getDatasetSubmission()->getStatus());
+                    $output->writeln('New submission ID in dataset is now: '
+                        . $dataset->getDatasetSubmission()->getId());
+
+                    //Use rabbitmq to process dataset file and persist the file details. This will
+                    //Trigger filer and hasher (via filer) to complete the process.
+                    $this->getContainer()->get('old_sound_rabbit_mq.dataset_submission_producer')->publish(
+                        $dataset->getDatasetSubmission()->getId(),
+                        'dataset.SFTP'
+                    );
+                }
+            } else {
+                throw new \Exception("Could not open $stubFileName, expected to be at same location as $infoFileName.");
+            }
+        } else {
+            throw new \Exception("Error: Could not open $infoFileName.");
+        }
+        return 0;
```

Comment:
Can remove this, Commands don't have to return anything. I know we've written few in the past like this.